File: globaldefensesurveillancetracking_ui/src/services/ActivityService.js
import http from "../http-common";

class ActivityService {
  getAllActivitys(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/activity/activitys`, searchDTO);
  }

  get(activityId) {
    return this.getRequest(`/activity/${activityId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/activity?field=${matchData}`, null);
  }

  addActivity(data) {
    return http.post("/activity/addActivity", data);
  }

  update(data) {
  	return http.post("/activity/updateActivity", data);
  }
  
  uploadImage(data,activityId) {
  	return http.postForm("/activity/uploadImage/"+activityId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ActivityService();

File: src/main/java/com/globaldefensesurveillancetracking/service/ActivityService.java
package com.globaldefensesurveillancetracking.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.globaldefensesurveillancetracking.domain.Activity;
import com.globaldefensesurveillancetracking.dto.ActivityDTO;
import com.globaldefensesurveillancetracking.dto.ActivitySearchDTO;
import com.globaldefensesurveillancetracking.dto.ActivityPageDTO;
import com.globaldefensesurveillancetracking.dto.ActivityConvertCriteriaDTO;
import com.globaldefensesurveillancetracking.service.GenericService;
import com.globaldefensesurveillancetracking.dto.common.RequestDTO;
import com.globaldefensesurveillancetracking.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ActivityService extends GenericService<Activity, Integer> {

	List<Activity> findAll();

	ResultDTO addActivity(ActivityDTO activityDTO, RequestDTO requestDTO);

	ResultDTO updateActivity(ActivityDTO activityDTO, RequestDTO requestDTO);

    Page<Activity> getAllActivitys(Pageable pageable);

    Page<Activity> getAllActivitys(Specification<Activity> spec, Pageable pageable);

	ResponseEntity<ActivityPageDTO> getActivitys(ActivitySearchDTO activitySearchDTO);
	
	List<ActivityDTO> convertActivitysToActivityDTOs(List<Activity> activitys, ActivityConvertCriteriaDTO convertCriteria);

	ActivityDTO getActivityDTOById(Integer activityId);







}





